### PR TITLE
create ci/cd pipeline to deploy react app to gh-pages 

### PR DESCRIPTION
**Pull request by hqtan**
fixes [this trello task](https://trello.com/c/9D5oqhLA/79-set-up-a-deployment-pipeline). 

ci/cd pipeline will auto-deploy latest changes of react app to gh-pages. 

changes introduced in this PR:
- `pipeline.yml` specifying the steps to automate the deployment of react app to gh-pages
- A workflow secret containing @hqtan's personal access token has been created to allow pipeline to deploy to gh-pages
- Had to update `npm run deploy` script to omit `gh-pages clean` command; it was causing the deployment to fail for some reason :shrug: 
- updated `.gitignore` to ignore `build/` folder (generated by `gh-pages` npm tool to build the react app)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "seechange",
   "version": "0.1.0",
   "private": true,
-  "homepage": "https://seechange.codeforaustralia.org",
+  "homepage": "https://hqtan.github.io/seechange",
```
Comment (klubiniecki) [inline package.json]: Hey! Is this a temporary change?

**Review by klubiniecki — APPROVED** [review]
@hqtan looks amazing, thanks! Just one comment. 🚀

File: package.json (hunk 1 above)
Comment (hqtan) [reply package.json]: oh crap... that was _meant_ to be a temporary change, when I was testing the pipeline in my fork!

I should've reverted it before making the PR! Thanks for spotting that @klubiniecki ! :bow:

**Comment by klubiniecki** [comment]
Sweet, let's get it out there in the wild and see what happens! 🟢 or 🔴 ? 😄

**Comment by klubiniecki** [comment]
Sweet, let's get it out there in the wild and see what happens! 🟢 or 🔴 ? 😄

**Pull request merged**